app/controllers/shelters_controller.rb

```diff
@@ -23,7 +27,12 @@ def edit
   def update
     shelter = Shelter.find(params[:id])
     shelter.update(shelter_params)
-    redirect_to "/shelters/#{shelter.id}"
+    if shelter.save
+      redirect_to "/shelters/#{shelter.id}"
+    else
+      flash[:error] = shelter.errors.full_messages.to_sentence
+      redirect_back(fallback_location: '/shelters')
```

Comment:
I think this can actually be `render: edit` vs having a redirect_back

Reply:
I tried render edit and it was not working for me. We can look in the am